docs/src/developing/clients/javascript-reference.md

```diff
@@ -0,0 +1,873 @@
+    'confirmed'
+);
+
+let airdropSignature = await connection.requestAirdrop(
+    fromPublicKey.publicKey,
+    web3.LAMPORTS_PER_SOL,
+);
+
+await connection.confirmTransaction(airdropSignature);
+
+let type = web3.SYSTEM_INSTRUCTION_LAYOUTS.Transfer;
+let data = Buffer.alloc(type.layout.span);
+let layoutFields = Object.assign({instruction: type.index});
+type.layout.encode(layoutFields, data);
+
+let recentBlockhash = await connection.getRecentBlockhash();
+
+let messageParams = {
+    accountKeys: [
+        fromPublicKey.publicKey.toString(),
+        toPublicKey.toString(),
+        web3.SystemProgram.programId.toString()
+    ],
+    header: {
+        numReadonlySignedAccounts: 0,
+        numReadonlyUnsignedAccounts: 1,
+        numRequiredSignatures: 1,
+    },
+    instructions: [
+        {
+        accounts: [0, 1],
+        data: bs58.encode(data),
+        programIdIndex: 2,
+        },
+    ],
+    recentBlockhash,
+};
+
+let message = new web3.Message(messageParams);
+
+let transaction = web3.Transaction.populate(
+    message,
+    [fromPublicKey.publicKey.toString()]
+);
+
+await web3.sendAndConfirmTransaction(connection, transaction, [fromPublicKey])
+```
+
+### NonceAccount
+
+[Source Documentation](https://solana-labs.github.io/solana-web3.js/classes/NonceAccount.html)
+
+Normally a transaction is rejected if a transaction's `recentBlockhash` field is too old. To provide for certain custodial services, Nonce Accounts are used.
+
+You can create a nonce account by first creating a normal account, then using `SystemProgram` to make the account a Nonce Account.
+
+#### Example Usage
+
+```javascript
+const web3 = require('@solana/web3.js');
+
+// Create connection
+let connection = new web3.Connection(
+    web3.clusterApiUrl('devnet'),
+    'confirmed',
+);
+
+// Generate accounts
+let account = web3.Keypair.generate();
+let nonceAccount = web3.Keypair.generate();
+
+// Fund account
+let airdropSignature = await connection.requestAirdrop(
+    account.publicKey,
+    web3.LAMPORTS_PER_SOL,
+);
+
+await connection.confirmTransaction(airdropSignature);
+
+// Get Minimum amount for rent exemption
+let minimumAmount = await connection.getMinimumBalanceForRentExemption(
+    web3.NONCE_ACCOUNT_LENGTH,
+);
+
+// Form CreateNonceAccount transaction
+let transaction = new web3.Transaction().add(
+web3.SystemProgram.createNonceAccount({
+    fromPubkey: account.publicKey,
+    noncePubkey: nonceAccount.publicKey,
+    authorizedPubkey: account.publicKey,
+    lamports: minimumAmount,
+}),
+);
+// Create Nonce Account
+await web3.sendAndConfirmTransaction(
+    connection,
+    transaction,
+    [account, nonceAccount]
+);
+
+let nonceAccountData = await connection.getNonce(
+    nonceAccount.publicKey,
+    'confirmed',
+);
+
+console.log(nonceAccountData);
+// NonceAccount {
+//   authorizedPubkey: PublicKey {
+//     _bn: <BN: 919981a5497e8f85c805547439ae59f607ea625b86b1138ea6e41a68ab8ee038>
+//   },
```

Comment:
Side note, it would be great if this logged in a prettier way. Do you mind writing an issue for that?

Reply:
Created #20022. I'll take a stab at it if it is still open tomorrow.